### PR TITLE
[terra-menu] Fixing padding issues on menu items

### LINKED ISSUE #1234: terra-menu Right Padding Non-existent
# Bug Report

## Description

We switched to using the terra-menu component in our project and I noticed that when I use `contentWidth="auto"`, there is no right padding. The text is crammed against the right side of the popup (see screenshot). Note, we are not using selection in our menu; on selection, we close the menu.
## Steps to Reproduce

1. Render a menu with a few options (it is easier to see if you have one of them longer than the other menu items)
2. Set the `contentWidth` prop to `auto`
3. Observe the popup is not sized properly to give the content room to breathe

## Additional Context / Screenshots

Screen Shot 2020-09-09 at 10 26 46

Here is (basically) my jsx code:
```
const filterOptions = ['value 1', 'value 2', 'a really long value 3', 'value 4'].map((type) => (
  <Menu.Item
    key={type}
    aria-label={type}
    text={type}
    onClick={(event) => {
      this.filterChanged(encounterType[type], event.target.textContent);
    }}
  />
));
```

## Expected Behavior

I would expect there to be a small amount of padding or margin to the right of the text.
## Possible Solution

Add some scss.
## Environment

* Component Name and Version: terra-menu - 6.40.0
* Browser Name and Version: Chrome - 85.0.4183.83
* Node/npm Version: Node 12/npm 6.13.4
* Webpack Version: 
* Operating System and version (desktop or mobile): mac OSX desktop

## @ Mentions

@dkasper-was-taken

### PR DESCRIPTION
**Pull request by jeremyfuksa**
Fixing the right margin on menu items, but only when they are not top-level nested menu items.

Signed-off-by: Jeremy Fuksa <hello@orangefla.me>

### Summary
This takes care of visual styling problems with menu items that have no end element injected by terra-arrange.

Closes #1234

### Deployment Link

https://terra-framew-menu-right-utytwk.herokuapp.com/components/terra-menu/menu/menu

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-menu/src/MenuItem.module.scss [hunk 1]
```diff
@@ -67,12 +67,17 @@
...
+    margin-right: var(--terra-menu-item-text-margin-right, 0);
+  }
+
   .text {
     color: var(--terra-menu-item-text-color, #1c1f21);
     font-size: var(--terra-menu-item-text-font-size, 13px);
     font-weight: var(--terra-menu-item-text-font-weight, normal);
     line-height: 1.538;
     margin-left: var(--terra-menu-item-text-margin-left, 10px);
+    margin-right: var(--terra-menu-item-text-margin-right, 24px);
```
Comment (jeremyfuksa) [inline packages/terra-menu/src/MenuItem.module.scss:80]: Is there a reason pixel values are being used here?

File: packages/terra-menu/src/clinical-lowlight-theme/MenuItem.module.scss [hunk 2]
```diff
@@ -19,6 +19,7 @@
     --terra-menu-item-padding: 0.286rem 0;
     --terra-menu-item-text-font-size: 13px;
     --terra-menu-item-text-font-weight: normal;
+    --terra-menu-item-text-margin-right: 0;
     --terra-menu-item-text-margin-left: 10px;
```
Comment (jeremyfuksa) [inline packages/terra-menu/src/clinical-lowlight-theme/MenuItem.module.scss:23]: Reason for pixel values?

File: packages/terra-menu/src/MenuItem.module.scss [hunk 3]
```diff
@@ -67,12 +67,17 @@
     }
   }
 
+  .is-top-level .text {
+    margin-right: var(--terra-menu-item-text-margin-right, 0);
```
Comment (mjhenkes) [inline packages/terra-menu/src/MenuItem.module.scss:71]: You'll need a unique theme variable here.

Comment (jeremyfuksa) [reply packages/terra-menu/src/MenuItem.module.scss]: Fixed with b912c82.

**Review by yuderekyu — APPROVED** [review]

**Review by mjhenkes — APPROVED** [review]

**Review by pranav300 — APPROVED** [review]

**Review by tbiethman — APPROVED** [review]

**Pull request merged**